source/acquaman/AMGenericStepScanConfiguration.cpp

```diff
@@ -0,0 +1,187 @@
+#include "AMGenericStepScanConfiguration.h"
+
+#include "ui/acquaman/AMGenericStepScanConfigurationView.h"
+#include "acquaman/AMGenericStepScanController.h"
+
+#include <math.h>
+
+AMGenericStepScanConfiguration::AMGenericStepScanConfiguration(QObject *parent)
+	: AMStepScanConfiguration(parent)
+{
+	setName("Generic Scan");
+	setUserScanName("Generic Scan");
+}
+
+AMGenericStepScanConfiguration::AMGenericStepScanConfiguration(const AMGenericStepScanConfiguration &original)
+	: AMStepScanConfiguration(original)
+{
+	setName(original.name());
+	setUserScanName(original.name());
+
+	computeTotalTime();
+
+	for (int i = 0, size = scanAxes_.count(); i < size; i++){
+
+		connect(scanAxisAt(i)->regionAt(0), SIGNAL(regionStartChanged(AMNumber)), this, SLOT(computeTotalTime()));
+		connect(scanAxisAt(i)->regionAt(0), SIGNAL(regionStepChanged(AMNumber)), this, SLOT(computeTotalTime()));
+		connect(scanAxisAt(i)->regionAt(0), SIGNAL(regionEndChanged(AMNumber)), this, SLOT(computeTotalTime()));
+		connect(scanAxisAt(i)->regionAt(0), SIGNAL(regionTimeChanged(AMNumber)), this, SLOT(computeTotalTime()));
+	}
+}
+
+AMGenericStepScanConfiguration::~AMGenericStepScanConfiguration(){}
+
+AMScanConfiguration *AMGenericStepScanConfiguration::createCopy() const
+{
+	AMScanConfiguration *configuration = new AMGenericStepScanConfiguration(*this);
+	configuration->dissociateFromDb(true);
+	return configuration;
+}
+
+AMScanController *AMGenericStepScanConfiguration::createController()
+{
+	AMScanActionController *controller = new AMGenericStepScanController(this);
+	controller->buildScanController();
+
+	return controller;
+}
+
+AMScanConfigurationView *AMGenericStepScanConfiguration::createView()
+{
+	return new AMGenericStepScanConfigurationView(this);
+}
+
+QString AMGenericStepScanConfiguration::technique() const
+{
+	return "Generic Scan";
+}
+
+QString AMGenericStepScanConfiguration::description() const
+{
+	return "Generic Scan";
+}
+
+QString AMGenericStepScanConfiguration::detailedDescription() const
+{
+	return "Does a generic scan over one or two controls with a variety of detectors.";
+}
+
+void AMGenericStepScanConfiguration::computeTotalTime()
+{
+	if (scanAxes_.count() == 1){
+
+		double size = fabs(double(scanAxes_.at(0)->regionAt(0)->regionEnd())-double(scanAxes_.at(0)->regionAt(0)->regionStart()));
+		int points = int((size)/double(scanAxes_.at(0)->regionAt(0)->regionStep()));
+
+		if ((size - (points + 0.01)*double(scanAxes_.at(0)->regionAt(0)->regionStep())) < 0)
+			points += 1;
+
+		else
+			points += 2;
+
```

Comment:
There is no calculation for `totalTime_`. Should we do it here? 
`totalTime_ = points *  (....->regionTime())`